### PR TITLE
Add function for NvAPI_GPU_GetVbiosVersionString

### PR DESCRIPTION
**Pull request by SveSop**
This grabs bios version from nvml for the graphics adapter.

### CODE REVIEW

File: src/sysinfo/nvapi_adapter.cpp [hunk 1]
```diff
@@ -209,4 +209,7 @@ namespace dxvk {
     nvmlReturn_t NvapiAdapter::NvmlDeviceGetUtilizationRates(nvmlUtilization_t *utilization) const {
         return m_nvml.DeviceGetUtilizationRates(m_nvmlDevice, utilization);
     }
+    nvmlReturn_t NvapiAdapter::NvmlDeviceGetVbiosVersion(char *version, unsigned int length) const {
```
Comment (jp7677) [inline src/sysinfo/nvapi_adapter.cpp:212]: Please add an empty line between methods.

File: src/sysinfo/nvml.cpp [hunk 2]
```diff
@@ -91,4 +97,7 @@ namespace dxvk {
     nvmlReturn_t Nvml::DeviceGetUtilizationRates(nvmlDevice_t device, nvmlUtilization_t *utilization) const {
         return m_nvmlDeviceGetUtilizationRates(device, utilization);
     }
-}
\ No newline at end of file
+    nvmlReturn_t Nvml::DeviceGetVbiosVersion(nvmlDevice_t device, char *version, unsigned int length) const {
```
Comment (jp7677) [inline src/sysinfo/nvml.cpp:100]: Please add an empty line between methods.

File: src/sysinfo/nvapi_adapter.h [hunk 3]
```diff
@@ -29,6 +29,7 @@ namespace dxvk {
         [[nodiscard]] std::string NvmlErrorString(nvmlReturn_t result) const;
         [[nodiscard]] nvmlReturn_t NvmlDeviceGetTemperature(nvmlTemperatureSensors_t sensorType, unsigned int* temp) const;
         [[nodiscard]] nvmlReturn_t NvmlDeviceGetUtilizationRates(nvmlUtilization_t* utilization) const;
+        [[nodiscard]] nvmlReturn_t NvmlDeviceGetVbiosVersion(char *version, unsigned int length) const;
```
Comment (jp7677) [inline src/sysinfo/nvapi_adapter.h:32]: Please bind the * to left side, thus `char* version`. I know, the styles are messy in this project, though I would prefer c++ style for our own types. Same for the method implementation.

**Review by jp7677 — CHANGES_REQUESTED** [review]

**Comment by SveSop** [comment]
Yeah, i suppose it is only a "one time info" kind of thing when it comes to vbios, not like frequency or stuff like that.

A wee bit confused about that `char *version` vs `char* version`, since i .. uhm.. kinda have a hard time knowing when it is "our own types"? Not being picky, but could not just all of them change, and just leave the official headers (nvapi.h/nvml.h) intact?

File: src/sysinfo/nvapi_adapter.h (hunk 3 above)
Comment (jp7677) [reply src/sysinfo/nvapi_adapter.h]: Please do that also for the implementation (in the cpp file.)

**Comment by jp7677** [comment]
> Yeah, i suppose it is only a "one time info" kind of thing when it comes to vbios, not like frequency or stuff like that.
> 
> A wee bit confused about that `char *version` vs `char* version`, since i .. uhm.. kinda have a hard time knowing when it is "our own types"? Not being picky, but could not just all of them change, and just leave the official headers (nvapi.h/nvml.h) intact?

I did that initially, but soon forgot that when adding a new method. So currently I keep directly copied method signatures as-is, thus "not our own", e.g. in the lop level nvapi_* files and the nvml.h/cpp since that is only a wrapper around nvml. Everything else should conform to having * on the left side. As I said, it's a bit messy :).

File: src/sysinfo/nvapi_adapter.cpp [hunk 4]
```diff
@@ -209,4 +209,8 @@ namespace dxvk {
     nvmlReturn_t NvapiAdapter::NvmlDeviceGetUtilizationRates(nvmlUtilization_t *utilization) const {
         return m_nvml.DeviceGetUtilizationRates(m_nvmlDevice, utilization);
     }
+
+    nvmlReturn_t NvapiAdapter::NvmlDeviceGetVbiosVersion(char *version, unsigned int length) const {
```
Comment (jp7677) [inline src/sysinfo/nvapi_adapter.cpp:213]: Could you please change the `*` here too?

**Review by jp7677 — CHANGES_REQUESTED** [review]

**Review by jp7677 — APPROVED** [review]

**Comment by jp7677** [comment]
Merged, thanks a lot @SveSop

**Pull request merged**